File: include/correlations/water-volume-factor.rb

begin_correlation("BwSpiveyMN", "WaterVolumeFactor", "RB_STB")
add_parameter("t", "Celsius", "Aqui Neylith",  "Quantity<Fahrenheit>(32)",  "Quantity<Fahrenheit>(530)") 
add_parameter("p", "mPascal", "Aqui Neylith",  "Quantity<psia>(14.5)",  "Quantity<psia>(29000)")
add_parameter("s", "Disolved_Salt_Percent", "Aqui Neylith",  0,  26)
add_author("Spivey et al.")
end_correlation()

################################################################

begin_correlation("BwMcCain", "WaterVolumeFactor", "RB_STB")
add_parameter("t", "Fahrenheit", "Aqui Neylith",  70,  250)
add_parameter("p", "psia", "Aqui Neylith",  1000,  5000)
add_author("McCain")
end_correlation()

################################################################

begin_correlation("BwMcCoy", "WaterVolumeFactor", "RB_STB")
add_parameter("t",  "Fahrenheit", "Aqui Neylith")
add_parameter("p",  "psia", "Aqui Neylith")
add_parameter("pb",  "psia", "Aqui Neylith")
add_parameter("s",  "Disolved_Salt_Percent", "Aqui Neylith")
add_parameter("cw",  "psia_1", "Aqui Neylith")
add_author("McCoy")
end_correlation()

File: include/correlations/water-volume-factor.cc
Quantity<RB_STB>
BwSpiveyMN__correlation__fct(const Quantity<Celsius>& t,
                             const Quantity<mPascal>& p,
                             const Quantity<Dissolved_Salt_Percent>& s)
{
  return BwSpiveyMN::get_instance()(t, p, s);
}

double
BwSpiveyMN__correlation__valid__fct(double t, double p, double s)
{
  return BwSpiveyMN::get_instance()(t, p, s).raw();
}

Quantity<RB_STB>
BwMcCain__correlation__fct(const Quantity<Fahrenheit>& t,
                           const Quantity<psia>& p)
{
  return BwMcCain::get_instance()(t, p);
}

double
BwMcCain__correlation__valid__fct(double t, double p)
{
  return BwMcCain::get_instance()(t, p).raw();
}

Quantity<RB_STB>
BwMcCoy__correlation__fct(const Quantity<Fahrenheit>& t,
                          const Quantity<psia>& p, const Quantity<psia>& pb,
                          const Quantity<Dissolved_Salt_Percent>& s,
                          const Quantity<psia_1>& cw)
{
  return BwMcCoy::get_instance()(t, p, pb, s, cw);
}

double
BwMcCoy__correlation__valid__fct(double t, double p, double pb, double s,
                                 double cw)
{
  return BwMcCoy::get_instance()(t, p, pb, s, cw).raw();
}
